features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/dashspend/PurchaseGiftCardFragmentV2.kt

```
@@ -0,0 +1,486 @@
+                    denominationQuantities.clear()
+                    viewModel.setIsFixedDenominationMultiple(isMultiple)
+                    viewModel.resetSelectedDenomination()
+                    // Keep current merchant; just switch UI mode
+                },
+                onToggleBalance = { showBalance = !showBalance },
+                onKeyInput = { key ->
+                    amountText = processAmountKeyInput(amountText, key)
+                    // Update viewModel order info so confirm dialog has up-to-date amount
+                    val fiatAmount = try {
+                        Fiat.parseFiat(Constants.USD_CURRENCY, amountText)
+                    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)